### PR TITLE
Add script to backfill Parent-Child level relationships

### PR DESCRIPTION
**Pull request by Hamms**
Follow-up to https://github.com/code-dot-org/code-dot-org/pull/41587 and https://github.com/code-dot-org/code-dot-org/pull/41547

Adds a one-off script we can run on each environment to populate the Parent-Child relationships of our new many-to-many table for existing levels.

Once this has been run on all environments, we can remove the fallback logic [here](https://github.com/code-dot-org/code-dot-org/blob/f316ffd2430b9d110b49961f997fb771bc469b29/dashboard/app/models/concerns/levels/levels_within_levels.rb#L123-L129) and [here](https://github.com/code-dot-org/code-dot-org/blob/f316ffd2430b9d110b49961f997fb771bc469b29/dashboard/app/models/concerns/levels/levels_within_levels.rb#L102-L111).

## Testing story

Tested by executing locally. Took 1m38.688s to run successfully; the five levels which reported errors were all confirmed to be the result of bad local data (levels whose `project_template_level_name` field contained placeholder values), and manual inspection of my local database after execution confirms that new relationships have been set up as desired.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by tim-dot-org — COMMENTED** [review]
how many levels does this affect on production? would it be worth running in batches, i.e. https://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches

**Comment by Hamms** [comment]
473 levels with contained levels and 3,636 levels with project templates; few enough that I'm not worried about batching for a script that's only going to need to be executed once. But I'd be more than willing to add batching if y'all think those numbers are indeed high enough.

**Review by tim-dot-org — APPROVED** [review]
oh nope that should be totally fine!

File: bin/oneoff/backfill_parent-child_relationships.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+require_relative '../../lib/cdo/only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+require_relative '../../dashboard/config/environment'
+
+def main
+  Level.includes(:child_levels).all.each do |level|
+    if level.contained_level_names.present?
+      begin
+        level.setup_contained_levels
```
Comment (maureensturgeon) [inline bin/oneoff/backfill_parent-child_relationships.rb:11]: I thought this migration was already performed with https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20210722193724_setup_contained_levels.rb, I'm curious why this is necessary.

**Review by maureensturgeon — APPROVED** [review]

File: bin/oneoff/backfill_parent-child_relationships.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+require_relative '../../lib/cdo/only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
```
Comment (sureshc) [inline bin/oneoff/backfill_parent-child_relationships.rb:3]: I think this script will be run manually on each managed environment? If it's not a scheduled script, we probably don't need the `only_one` logic.

File: bin/oneoff/backfill_parent-child_relationships.rb (hunk 1 above)
Comment (Hamms) [reply bin/oneoff/backfill_parent-child_relationships.rb:11]: Good catch! Yes, this migration was run on all environments with that change, but there was some bad data in a couple of our environments and also a small bug, both of which mean that there is a subset of levels that didn't get correctly updated by that. So this script is pulling double duty; both migrating over the project template levels for the first time and finishing the migration for the contained levels.

File: bin/oneoff/backfill_parent-child_relationships.rb (hunk 2 above)
Comment (Hamms) [reply bin/oneoff/backfill_parent-child_relationships.rb:3]: that's correct, but it takes a couple minutes to run so I left this in just for safety. Is there any downside to including this?

**Pull request merged**